developing-packages.md

```diff
@@ -0,0 +1,174 @@
+---
+layout: page
+title: Developing Packages & Plugins
+permalink: /developing-packages/
+---
+
+* TOC
+{:toc}
+
+## Package introduction
+
+Dart packages enable the creation of modular code that can easily be shared. A
+minimal package with a single library consists of:
+
+* A `pubspec.yaml` file: A metadata file that declared the package name,
+  version, author, etc.
+
+* A `lib` directory containing the public code in the package, minimally a
+  single `<package-name>.dart` file.
+
+For additional details, see the [Dart library package](https://www.dartlang.org/guides/libraries/create-library-packages) documentation.
+
+### Package types {#types}
+
+Flutter supports several kinds of packages:
+
+* *Dart packages*: General packages written in Dart without any dependencies on
+  any specific frameworks. They can be used across many frameworks such as
+  [AngularDart](https://webdev.dartlang.org/angular) and Flutter. A concrete
+  example is the [`path`](https://pub.dartlang.org/packages/path) package.
+
+* *Flutter packages*: General packages which contain Flutter specific
+  functionality and thus have a dependency on the Flutter framework, restricting
+  their use to Flutter only. A concrete example is the
+  [`fluro`](https://pub.dartlang.org/packages/fluro) package.
+
+* *Flutter plugins*: Flutter packages which in additional to the Dart code also
+  contain a platform-specific implementation for Android, for iOS, or for both.
+  A concrete example is the
+  [`battery`](https://pub.dartlang.org/packages/battery) plugin.
+
+## Developing packages
+
+### Create a package project
+
+Flutter currently lacks a dedicated command for creating a package project
+(issue [#10377](https://github.com/flutter/flutter/issues/10377)). As a
+workaround use these steps to create a package called `example`:
+
+1. Create a standard Flutter application:
+    * `flutter create example`
+
+1. Delete the iOS and Android application containers:
+  * `rm -rf ios`
+  * `rm -rf android`
+
+
+1. Remove the main application:
+  * `rm lib/main.dart`
+
+1. Add a main package file `lib/example.dart` with the following contents:
+    ``` dart
+    library example;
+
+    /// Add public package code (Classes, etc.) here:
+    ```  
+
+## Developing plugins: platform-specific packages {#plugin}
+
+Packages enable sharing Dart code. If you want to develop packages that call
```

Comment:
Packages enable sharing code. I wouldn't qualify it to "Dart code" here. That makes it easier later to explain a plugin.